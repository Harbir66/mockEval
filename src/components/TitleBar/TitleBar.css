.title-bar {
  flex: none;
  color: #000;
  display: flex;
  justify-content: space-between;
  align-items: center;
  font-size: 2rem;
  font-weight: 600;
  width: 100%;
  padding: 0 10%;
}

.title-bar button {
  background: none;
  border: none;
  color: #000;
  font-size: 1.5rem;
  font-weight: 600;
  cursor: pointer;
}

@media (max-width: 455px) {
  .title-bar {
    width: 455px;
    padding: 0;
  }
  .title-bar h1 {
    margin-left: 30px;
  }
  .title-bar button {
    margin-right: 30px;
  }
}
