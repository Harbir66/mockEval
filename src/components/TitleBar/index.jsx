import React from 'react';
import './TitleBar.css';
import PropTypes from 'prop-types';
import iconGrid from '../../assets/icon-grid.svg';
import iconGenre from '../../assets/icon-genre.svg';

function TitleBar({ toggle, handleToggle }) {
  const title = toggle ? 'genres' : 'all songs';
  const icon = toggle ? iconGrid : iconGenre;
  return (
    <div className="title-bar">
      <h1>{title}</h1>
      <button type="button" onClick={handleToggle}>
        <img src={icon} alt="Icon" />
      </button>
    </div>
  );
}

export default TitleBar;

TitleBar.propTypes = {
  toggle: PropTypes.bool.isRequired,
  handleToggle: PropTypes.func.isRequired,
};
