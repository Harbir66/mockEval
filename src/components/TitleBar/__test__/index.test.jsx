import React from 'react';
import { render } from '@testing-library/react';
import TitleBar from '../index';

describe('TitleBar', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <TitleBar toggle={false} handleToggle={jest.fn()} />
    );
    expect(asFragment).toMatchSnapshot();
  });
});
