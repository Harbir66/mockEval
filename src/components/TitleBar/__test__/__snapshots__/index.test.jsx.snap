// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBar should render correctly 1`] = `[Function]`;
